diff-hl-flydiff.el

```diff
@@ -0,0 +1,203 @@
+
+    (advice-add 'vc-git-working-revision :override
+      #'diff-hl-flydiff/vc-git-working-revision)
+    (advice-add 'vc-git-mode-line-string :override
+      #'diff-hl-flydiff/vc-git-mode-line-string)))
+
+(defun diff-hl-flydiff-make-temp-file-name (file rev &optional manual)
+  "Return a backup file name for REV or the current version of FILE.
+If MANUAL is non-nil it means that a name for backups created by
+the user should be returned."
+  (let* ((auto-save-file-name-transforms
+           `((".*" ,temporary-file-directory t))))
+    (expand-file-name
+      (concat (make-auto-save-file-name)
+        ".~" (subst-char-in-string
+               ?/ ?_ rev)
+        (unless manual ".") "~")
+      temporary-file-directory)))
+
+(defun diff-hl-flydiff-create-revision (file revision)
+  "Read REVISION of FILE into a buffer and return the buffer."
+  (let ((automatic-backup (diff-hl-flydiff-make-temp-file-name file revision))
+         (filebuf (get-file-buffer file))
+         (filename (diff-hl-flydiff-make-temp-file-name file revision 'manual)))
+    (unless (file-exists-p filename)
+      (if (file-exists-p automatic-backup)
+        (rename-file automatic-backup filename nil)
+        (with-current-buffer filebuf
+          (let ((failed t)
+                 (coding-system-for-read 'no-conversion)
+                 (coding-system-for-write 'no-conversion))
+            (unwind-protect
+              (with-temp-file filename
+                (let ((outbuf (current-buffer)))
+                  ;; Change buffer to get local value of
+                  ;; vc-checkout-switches.
+                  (with-current-buffer filebuf
+                    (vc-call find-revision file revision outbuf))))
+              (setq failed nil)
+              (when (and failed (file-exists-p filename))
+                (delete-file filename)))))))
+    filename))
+
+(defun diff-hl-flydiff-buffer-with-head ()
+  "View the differences between BUFFER and its associated file.
+This requires the external program `diff' to be in your `exec-path'."
+  (interactive)
+  (vc-ensure-vc-buffer)
+  (with-current-buffer (get-buffer (current-buffer))
+    (let ((rev (diff-hl-flydiff-create-revision
+                 buffer-file-name
+                 (vc-working-revision buffer-file-name
+                   (vc-responsible-backend buffer-file-name))))
+           (temporary-file-directory
+             (if (file-directory-p "/dev/shm/")
+               "/dev/shm/"
+               temporary-file-directory)))
+      (diff-no-select rev (current-buffer) "-U 0" 'noasync
+        (get-buffer-create " *diff-hl-diff*")))))
+
+
+(defun diff-hl-flydiff/update (old-fun &optional auto)
+  (unless (and auto
+            (or
+              (= diff-hl-flydiff-modified-tick (buffer-modified-tick))
+              (file-remote-p default-directory)
+              (not (buffer-modified-p))))
+    (funcall old-fun)))
+
+(defun diff-hl-flydiff/changes (&rest args)
```

Comment:
No, because the method of actually getting the changes is different. The new method is a superset of the old one, though, so you might consider just making it the default for the regular `diff-hl-mode`. Then I can just use a simple predicate list here.